node/core/dispute-coordinator/src/participation/queues/mod.rs

```diff
@@ -231,15 +226,31 @@ impl Queues {
 		&mut self,
 		comparator: CandidateComparator,
 		priority: ParticipationPriority,
-		req: ParticipationRequest,
+		mut req: ParticipationRequest,
 	) -> std::result::Result<(), QueueError> {
 		if priority.is_priority() {
 			if self.priority.len() >= PRIORITY_QUEUE_SIZE {
 				return Err(QueueError::PriorityFull)
 			}
-			// Remove any best effort entry:
-			self.best_effort.remove(&comparator);
-			self.priority.insert(comparator, req);
+			// Remove any best effort entry, using it to replace our new
+			// request. This preserves the original request timer from
+			// the older request.
```

Comment:
nit: I find this function hard to follow now, with all the nested ifs and elses. It would help to have a docstring for the function. Also, while the comments are awesome (⭐️), they are a bit repetitive, and maybe can be moved to the docstring. i.e. I envision something like:

```
/// # Request timers
///
/// [`ParticipationRequest`]s contain request timers. 
/// Where an old request would be replaced by a new one, we try to keep the old request.
/// This prevents request timers from resetting on each new request.
```

so the comment here becomes

```suggestion
			// Remove any best effort entry, using it to replace our new
			// request.
```